### ISSUE TITLE
Implement Into<learning::Error> for rulinalg errors

### ISSUE THREAD

**Issue by AtheMathmo** [AtheMathmo/rusty-machine#144]
This will make it easier for us to return errors caused by rulinalg failures - right now we tend to do something like: `try!(rulinalg_result.map_err(|_| { ... }))`. We could shorten this to: `try!(rulinalg_result.into())`.

This will need some changes on the rulinalg side to expose more of the `linalg::Error`. This probably just involves adding a new function to return the `error` field.

**Comment by AtheMathmo** [comment]
We have `From` implemented thanks to #142 (thanks @andrewcsmith ). Is this enough?

**Comment by andrewcsmith** [comment]
I'm not sure if this is actually enough...I think there are other errors that will need to be converted. Should be able to take a look soon though and compare it to what I've done with a different rpoject. But I totally agree with this issue – it would be great to allow end-users to access / recover from problems in the process.

**Comment by AtheMathmo** [comment]
I'm going to close this off and we can reopen later if an issue arises. I think that the `From` implementation is enough right now to get us by.
